### DISCUSSION TITLE
Screensaver: durations >60s

### DISCUSSION THREAD

**Discussion by AaronJowitt** [PE5PVB/TEF6686_ESP32#277] [Ideas]
A few long options would be useful. 30 minutes, 1 hour.

**Comment by andimik** [comment]
You can change the sources if you want. Therefore, this is not an issue.

https://github.com/PE5PVB/TEF6686_ESP32/blob/e605fc2e5a732add5925fc4cb72bf321d27bb92e/TEF6686_ESP32.ino#L159

**Comment by Overland-DX** [comment]
@AaronJowitt 
I agree that we should have more choices, so we could really use it as a sleep timer. But as andimik says, this works as intended.

@andimik 
I've played around with that part a bit, but I could never get the value to be higher than 200 sec. So I think there are some limitations there.
If you know of a solution to choose 30 minutes or more, I'm interested in knowing how.

**Comment by andimik** [comment]
But was is the intention to have such a long time?

**Comment by Overland-DX** [comment]
I use the radio on the bedside table when I go to bed. Many radios have this function where they switch off automatically after a specified time.
This radio also has this option called "Deep sleep", but becomes a bit pointless in this setting when you can only choose a maximum of 60 seconds.
It would be nice to have this option on this radio too if it is possible. But the radio uses little power anyway, so it's not that important :)

**Comment by andimik** [comment]
Have you tried 255 sec?

**Comment by Overland-DX** [comment]
![image](https://github.com/PE5PVB/TEF6686_ESP32/assets/102610897/2e55805c-2ec4-45ff-9b95-f707a3658706)
It seems like the max is 255 for some reason.
Could there be a way to increase that value?

**Comment by andimik** [comment]
Yes, you need to replace `byte`. But I am not a programmer...

**Comment by Overland-DX** [comment]
I'm certainly not a programmer either, but I like to explore a bit.
I was able to change the value from seconds to minutes. Haven't tested much, but a minute seemed fine, testing 10 min now.

File: constants.h 
line: 15
Changed from:
#define TIMER_SCREENSAVER_BASE     1000000
to:
#define TIMER_SCREENSAVER_BASE     60000000

updated: 10 mins test works fine :)

**Comment by andimik** [comment]
But again, what is the benefit of such a long time? The screen goes off, but sound is still available.

I am using 3 seconds in case the screen makes too much noise on weak signals.

But I like the idea having another menu option `sec` resp. `min` switching between these two values for `TIMER_SCREENSAVER_BASE`

**Comment by Overland-DX** [comment]
Yes, it is very useful the way it works in the first place.
I tested a bit more yesterday and it looked like everything worked fine up to 30 minutes, but when I tried 50, 60 and 70 minutes, the radio became unstable and it went into a restart loop. So it's probably not the right way to do it.

**Comment by Overland-DX** [comment]
@AaronJowitt

**New Timeout-timer function in latest version.**

~The only simple solution I can suggest is to change the value from seconds to minutes.~
~You need to change some values in at least one file to make this happen.~

~First you need to change this line in constants.h:~
~from:~
~#define TIMER_SCREENSAVER_BASE 1000000~
~to:~
~#define TIMER_SCREENSAVER_BASE 60000000UL~

~In order for it to appear correctly on the display, I recommend changing this in language.h from seconds to minutes.:~
~from:~
~"sec.", // 92~
~to:~
~"min.", // 92~

~I don't know how correct this solution is, but testing so far works fine on my radio.~
~Just keep in mind that the shortest value will then be 1 minute and the longest I have got to work is 60 minutes.~

**Comment by ohmytime** [comment]
The premise is: Tunner can sleep only when the power option is Deep Sleep.

As user's power options is screen off now, if the timer expires after 30 minutes, should we change it to Deep sleep?
Therefore, scheduled sleep is not impossible. The problem is that if the user's setting is not Deep Sleep, we may need to change the user's setting.

An option is to set a separate sleep option.

**Reply by Overland-DX** [reply]
A separate deep sleep option would be perfect. It would perhaps fit best in the main settings menu?
I have tried to separate the screensaver section with a view to creating a separate deep sleep option, but have so far been unsuccessful.
The option to use screensaver at the same time as the radio switches off after an hour or so would save a lot on battery capacity.

**Comment by PE5PVB** [comment]
I've added a seperate time-out timer. You can choose for off, 15m, 30m, 60m and 90m.
Timer will reset everytime you use a button or the rotary.
Radio will go in full deep-sleep, including the processor.
To wake up, just jiggle the knob.

**Reply by Overland-DX** [reply]
It's incredibly amazing. Thank you very much! It will be a useful feature for me and hopefully for many others as well.
